### PR TITLE
feat(cli): Manage Vulnerability Exceptions

### PR DESCRIPTION
**Pull request by dmurray-lacework**
## Summary

Manage Vulnerability Exceptions in the CLI

```
lacework vulnerability-exceptions create
lacework vulnerability-exceptions list
lacework vulnerability-exceptions show <id>
lacework vulnerability-exceptions delete <id>
```

## How did you test this change?

run `lacework vulnerability-exceptions create | list | show | delete`
add unit test `cli/cmd/vulnerability_exceptions_test.go`

## Issue

https://lacework.atlassian.net/browse/ALLY-752

### CODE REVIEW

File: cli/cmd/vulnerabilty_exceptions.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+			if cli.JSONOutput() {
+				return cli.OutputJSON(vulnerabilityExceptions)
+			}
+
+			var rows [][]string
+			for _, vuln := range vulnerabilityExceptions.Data {
+				rows = append(rows, []string{vuln.Guid, vuln.ExceptionName, vuln.ExceptionType, vuln.Status()})
+			}
+
+			cli.OutputHuman(renderSimpleTable([]string{"GUID", "NAME", "TYPE", "ENABLED"}, rows))
```
Comment (afiune) [inline cli/cmd/vulnerabilty_exceptions.go:81]:
```suggestion
			cli.OutputHuman(renderSimpleTable([]string{"GUID", "NAME", "TYPE", "STATE"}, rows))
```

**Review by afiune — CHANGES_REQUESTED** [review]
Tested the create with odd values and got a few panics 😢 

Screen Shot 2022-01-26 at 8 19 07 PM

Screen Shot 2022-01-26 at 8 20 40 PM

Also when I tried to list a vulnerability exception..... 😞 
Screen Shot 2022-01-26 at 8 22 11 PM

File: cli/cmd/vulnerabilty_exceptions.go [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+			if cli.JSONOutput() {
+				return cli.OutputJSON(vulnerabilityExceptions)
+			}
+
+			var rows [][]string
+			for _, vuln := range vulnerabilityExceptions.Data {
+				rows = append(rows, []string{vuln.Guid, vuln.ExceptionName, vuln.ExceptionType, vuln.Status()})
+			}
+
+			cli.OutputHuman(renderSimpleTable([]string{"GUID", "NAME", "TYPE", "STATE"}, rows))
```
Comment (afiune) [inline cli/cmd/vulnerabilty_exceptions.go:83]: This tables looks better (as we talked about):
```suggestion
			cli.OutputHuman(renderCustomTable([]string{"GUID", "NAME", "TYPE", "STATE"}, rows,
				tableFunc(func(t *tablewriter.Table) {
					t.SetBorder(false)
					t.SetAutoWrapText(false)
				})))
```

File: cli/cmd/vulnerabilty_exceptions.go [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+			return nil
+		},
+	}
+	// show command is used to retrieve a lacework vulnerability exception by id
+	vulnerabilityExceptionShowCommand = &cobra.Command{
+		Use:   "show",
+		Short: "Get vulnerability exception by id",
+		Long:  "Get a single vulnerability exception by it's vulnerability exception ID.",
```
Comment (afiune) [inline cli/cmd/vulnerabilty_exceptions.go:91]: In some docs you refer to id as `id` and some other places as `ID`.

Which one should we use? 🤔 

```
➜  go-sdk git:(dmurray-lacework/ALLY-752) ✗ git grep Short | grep -i " id"
cli/cmd/alert_rules.go:		Short: "Show an alert rule by id",
cli/cmd/report_rules.go:		Short: "Show a report rule by id",
cli/cmd/resource_groups.go:		Short: "Get resource group by id",
cli/cmd/vuln_host.go:		Short: "List the hosts that contain a specified CVE id in your environment",
cli/cmd/vulnerabilty_exceptions.go:		Short: "Get vulnerability exception by id",
➜  go-sdk git:(dmurray-lacework/ALLY-752) ✗ git grep Long | grep -i " id"
cli/cmd/alert_rules.go:		Long:  "Show a single alert rule by it's ID.",
cli/cmd/alert_rules.go:		Long:  "Delete a single alert rule by it's ID.",
cli/cmd/compliance_azure.go:		Long: `List all Azure subscriptions from the provided Tenant ID.
cli/cmd/compliance_gcp.go:		Long:  `List all GCP projects and organization IDs.`,
cli/cmd/compliance_gcp.go:		Long: `List all GCP projects from the provided organization ID.
cli/cmd/lql_delete.go:		Long: `Delete a single LQL query by providing the query id.
cli/cmd/policy.go:		Long: `Delete a policy by providing the policy id.
cli/cmd/report_rules.go:		Long:  "Show a single report rule by it's id.",
cli/cmd/report_rules.go:		Long:  "Delete a single report rule by it's ID.",
cli/cmd/resource_groups.go:		Long:  "Get a single resource group by it's resource group ID.",
cli/cmd/resource_groups.go:		Long:  "Delete a single resource group by it's resource group ID.",
cli/cmd/vuln_host.go:		Long: `List the hosts that contain a specified CVE id in your environment.
cli/cmd/vulnerabilty_exceptions.go:		Long:  "Get a single vulnerability exception by it's vulnerability exception ID.",
cli/cmd/vulnerabilty_exceptions.go:		Long:  "Delete a single vulnerability exception by it's vulnerability exception ID.",```

File: cli/cmd/vulnerabilty_exceptions.go [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+		return createHostVulnerabilityException()
+	case "Container":
+		return createContainerVulnerabilityException()
+	default:
+		return errors.New("unknown vulnerability exception type")
+	}
+}
+
+func vulnerabilityExceptionFixableEnabled(fixable []int) string {
+	if fixable == nil {
```
Comment (afiune) [inline cli/cmd/vulnerabilty_exceptions.go:247]: I think you also need to check if fixable has at least one element, if not, we will panic
```suggestion
	if fixable == nil || len(fixable) == 0 {
```

File: cli/cmd/vulnerabilty_exceptions.go [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+			if !cli.InteractiveMode() {
+				return errors.New("interactive mode is disabled")
+			}
+
+			err := promptCreateVulnerabilityException()
+			if err != nil {
+				return errors.Wrap(err, "unable to create vulnerability exception")
+			}
+
+			cli.OutputHuman("The vulnerability exception was created.\n")
```
Comment (afiune) [inline cli/cmd/vulnerabilty_exceptions.go:145]: Should we print the GUID here? again, some commands print it and some other don't. (inconsistencies)

What is our standard?

```
cli/cmd/alert_rules.go:			cli.OutputHuman(fmt.Sprintf("The alert rule was created with GUID %s \n", response.Data.Guid))
cli/cmd/integration.go:			cli.OutputHuman("The integration was created.\n")
cli/cmd/lql_create.go:	cli.OutputHuman("The query %s was created.\n", create.Data.QueryID)
cli/cmd/policy_create.go:	cli.OutputHuman(fmt.Sprintf("The policy %s was created.\n", createResponse.Data.PolicyID))
cli/cmd/report_rules.go:			cli.OutputHuman("The report rule was created with GUID %s\n", response.Data.Guid)
cli/cmd/resource_groups.go:			cli.OutputHuman("The resource group was created.\n")
cli/cmd/vulnerabilty_exceptions.go:			cli.OutputHuman("The vulnerability exception was created.\n")
```

File: cli/cmd/vulnerabilty_exceptions.go (hunk 3 above)
Comment (dmurray-lacework) [reply cli/cmd/vulnerabilty_exceptions.go:91]: I prefer `ID`

File: cli/cmd/vulnerabilty_exceptions.go (hunk 5 above)
Comment (dmurray-lacework) [reply cli/cmd/vulnerabilty_exceptions.go:145]: I think it is useful to pass back the ID where we can

**Review by afiune — APPROVED** [review]
tenor-122420412

**Pull request merged**